### PR TITLE
cleaning up biosample and participant

### PR DESCRIPTION
**Pull request by satra**
this is a draft PR to try and clean up the asset model. i haven't updated tests yet. putting the model changes here up for discussion.

### CODE REVIEW

File: dandi/models.py [hunk 1]
```diff
@@ -396,9 +396,29 @@ class BioSample(DandiBaseModel):
     )
     anatomy: Optional[List[Anatomy]] = Field(
         None,
-        description="UBERON based identifier for the location of the sample",
+        description="UBERON based identifier for what organ the sample belongs "
+        "to. Use the most specific descriptor.",
         nskey="dandi",
     )
+
+    wasDerivedFrom: Optional[List["BioSample"]] = Field(None, nskey="prov")
```
Comment (satra) [inline dandi/models.py:404]: this recursive formulation should allow us to have cell ids, slice ids, and tissue ids.

File: dandi/models.py [hunk 2]
```diff
@@ -396,9 +396,29 @@ class BioSample(DandiBaseModel):
...
+        "rdfs:label": "Information about the biosample.",
+        "nskey": "dandi",
+    }
+
+
+BioSample.update_forward_refs()
+
+
+class Participant(DandiBaseModel):
+    """Description about the sample that was studied"""
```
Comment (satra) [inline dandi/models.py:417]: separating out participant details

File: dandi/models.py [hunk 3]
```diff
@@ -632,6 +652,9 @@ class AssetMeta(CommonModel):
     )
 
     wasDerivedFrom: Optional[List[BioSample]] = Field(None, nskey="prov")
+    wasAttributedTo: List[Participant] = Field(
+        None, description="Participant(s) to which this file belongs to", nskey="prov"
+    )
```
Comment (satra) [inline dandi/models.py:657]: a given asset may have multiple participants potentially. how we deal with this in the name is something we will have to figure out.

**Comment by yarikoptic** [comment]
**now** master should be 3.9 testing friendly , so the only issues are of the `lint`

<details>
<summary>lint errors</summary> 

```shell
Run tox -e lint
lint create: /home/runner/work/dandi-cli/dandi-cli/.tox/lint
lint installdeps: flake8
lint installed: flake8==3.8.4,importlib-metadata==3.3.0,mccabe==0.6.1,pycodestyle==2.6.0,pyflakes==2.2.0,typing-extensions==3.7.4.3,zipp==3.4.0
lint run-test-pre: PYTHONHASHSEED='1970909900'
lint run-test: commands[0] | flake8 --config=setup.cfg dandi setup.py
dandi/tests/test_metadata.py:592:101: E501 line too long (132 > 100 characters)
dandi/tests/test_metadata.py:638:101: E501 line too long (109 > 100 characters)
dandi/tests/test_metadata.py:673:101: E501 line too long (221 > 100 characters)
dandi/tests/test_metadata.py:741:101: E501 line too long (109 > 100 characters)
dandi/tests/test_metadata.py:766:101: E501 line too long (109 > 100 characters)
dandi/tests/test_metadata.py:787:101: E501 line too long (106 > 100 characters)
dandi/tests/test_metadata.py:808:101: E501 line too long (106 > 100 characters)
dandi/tests/test_metadata.py:834:101: E501 line too long (131 > 100 characters)
dandi/tests/test_metadata.py:870:101: E501 line too long (221 > 100 characters)
dandi/tests/test_metadata.py:880:101: E501 line too long (133 > 100 characters)
dandi/tests/test_metadata.py:901:101: E501 line too long (119 > 100 characters)
dandi/tests/test_metadata.py:942:101: E501 line too long (119 > 100 characters)
dandi/tests/test_metadata.py:963:101: E501 line too long (109 > 100 characters)
dandi/tests/test_metadata.py:1002:101: E501 line too long (109 > 100 characters)
dandi/tests/test_metadata.py:1012:101: E501 line too long (106 > 100 characters)
dandi/tests/test_metadata.py:1022:101: E501 line too long (133 > 100 characters)
dandi/tests/test_metadata.py:1032:101: E501 line too long (127 > 100 characters)
dandi/tests/test_metadata.py:1042:101: E501 line too long (155 > 100 characters)
dandi/tests/test_metadata.py:1097:101: E501 line too long (102 > 100 characters)
dandi/tests/test_metadata.py:1127:101: E501 line too long (1147 > 100 characters)
dandi/tests/test_metadata.py:1149:101: E501 line too long (124 > 100 characters)
ERROR: InvocationError for command /home/runner/work/dandi-cli/dandi-cli/.tox/lint/bin/flake8 --config=setup.cfg dandi setup.py (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   lint: commands failed
Error: Process completed with exit code 1.
```
</details>

aren't you using pre-commit?!

**Comment by satra** [comment]
pre-commit isn't catching these linting things. do remember pre-commit uses black which is opinionated.

**Comment by yarikoptic** [comment]
hm, I just remember it splitting long lines for me and staying largely inline with what we have setup for linters... odd

**Comment by yarikoptic** [comment]
Dear @jwodder , could you please finalize this PR (if I got it right -- only linting is remaining) so we could merge it (@satra I believe is already holidaying officially so we should not pester him to not ruin that important nowadays activity).

**Comment by jwodder** [comment]
@yarikoptic Delinted.

**Comment by satra** [comment]
@yarikoptic - the tests that fail here are the same tests that fail on master and unrelated to this PR.

**Comment by yarikoptic** [comment]
hm, something changed in API server I guess -- #283 which had dealt with new code for API was all green when merged 2 days ago.  so indeed let's just proceed with merge and https://github.com/dandi/dandi-cli/issues/322 is now dedicated to test failure

**Pull request merged**